rholang/src/main/scala/coop/rchain/rholang/interpreter/error/MLock.scala

```diff
@@ -0,0 +1,41 @@
+package coop.rchain.rholang.interpreter.error
+
+import cats.effect.{Bracket, Concurrent}
+import cats.implicits._
+import cats.effect.concurrent.MVar
+
+final class MLock[F[_]](lock: MVar[F, Unit])(implicit bracketF: Bracket[F, Throwable]) {
+
+  def acquire: F[Unit] = lock.take
```

Comment:
I'm not happy with the interface of MLock eposing low-level primitives like aquire and release. Can't we just expose resource?

Reply:
The `withLock` function provides something similar to resource. It's not that simple when you want to use `MLock` as a primitive in more complex locking structures. See the subsequent PR.

Reply:
then we should replace withLock with a proper Resource that is composable with other resources.

And we should also hide acquire/release because they are error prone